src/API/Site/Controllers/MerchantCenter/ShippingRateController.php

```diff
@@ -186,14 +186,11 @@ protected function get_create_rate_callback(): callable {
 				$existing       = ! empty( $existing_query->get_results() );
 
 				if ( $existing ) {
-					$update_query->update(
-						$data,
-						[
-							'id' => $existing_query->get_results()[0]['id'],
-						]
-					);
+					$rate_id = $existing_query->get_results()[0]['id'];
+					$update_query->update( $data, [ 'id' => $rate_id ] );
 				} else {
 					$update_query->insert( $data );
+					$rate_id = $update_query->last_insert_id();
```

Comment:
❓ 

I guess that the probability that another plugin or function inserts data at the same time is low and therefore it is safe to use the last_insert_id?

Reply:
As far as I can see, that wouldn't be possible, because some other code would be running in a separate PHP process which would have a different instance of $wpdb. The $wpdb->insert_id is saved after the insert call so it's not retrieved from DB at a later state.

If we want we can store the last insert ID in the Query class (just in case they aren't called in succession). But I indicated in the function "Must be called after insert". Do you think I should stress that it should be called right after, or do you think it's better to add a safeguard by storing a copy of it in the Query class?

Reply:
I went with storing a local copy as an extra precaution, see 84b2d24